libraries/AP_Terrain/AP_Terrain.cpp

```diff
@@ -143,22 +147,30 @@ bool AP_Terrain::height_amsl(const Location &loc, float &height)
         home_loc = loc;
     }
 
+    // apply correction which assumes home altitude is at terrain altitude
+    if (corrected) {
+        height += (ahrs.get_home().alt * 0.01f) - home_height;
+    }
```

Comment:
Thanks OXNIARF, I think you've found a bug.  I think height_terrain_difference_home should also be setting that to true.  What's going on here is we are assuming that the terrain altitude is zero at home (i.e. that the vehicle is taking off from the terrain).  The way we do that, for better or worse, is we essentially adjust the terrain up and down.